src/main/java/frc/robot/subsystems/swerve/swervestatehelpers/AimAssistUtils.java

```diff
@@ -0,0 +1,50 @@
+package frc.robot.subsystems.swerve.swervestatehelpers;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import frc.robot.subsystems.swerve.SwerveConstants;
+import frc.robot.subsystems.swerve.SwerveMath;
+
+import java.util.function.Supplier;
+
+import static frc.robot.subsystems.swerve.SwerveMath.getDriveMagnitude;
+
+public class AimAssistUtils {
+
+    /**
+     * @param currentSpeeds current chassis speeds, field relative
+     * @param inputSpeeds the speeds that the joysticks command on the robot
+     * @param robotRotationSupplier supplier of the robot rotation
+     * @param targetRotationSupplier target angle for the rotation (as a supplier)
+     * @param swerveConstants the constants of the swerve
+     *
+     * @return the aim assisted chassis speeds with only rotation speeds assisted (to look at somewhere)
+     *
+     * */
+
+    public static ChassisSpeeds getRotationAssistedSpeeds(ChassisSpeeds inputSpeeds,
+            ChassisSpeeds currentSpeeds,
+            Supplier<Rotation2d> robotRotationSupplier,
+            Supplier<Rotation2d> targetRotationSupplier,
```

Comment:
why supplier?

Reply:
if the target angle is changing, like when looking at target and it moves. you are right, it can be just rotation 2d, but I prefer passing it as a supplier.

Reply:
but the method gets called periodically

Reply:
so you dont need supplier